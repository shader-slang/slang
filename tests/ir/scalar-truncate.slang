//TEST(compute):COMPARE_COMPUTE_EX:-slang -compute -shaderobj
//TEST(compute, vulkan):COMPARE_COMPUTE_EX:-vk -compute -shaderobj

//TEST_INPUT:ubuffer(data=[0 0 0 0], stride=4):out,name=outputBuffer
RWStructuredBuffer<uint> outputBuffer;

[numthreads(1, 1, 1)]
void computeMain(uint3 dispatchThreadID : SV_DispatchThreadID)
{
     {
        int16_t a = 30000;      // 0x00007530 -> 0x7530 (30000) in 16-bit.
        int16_t b = -200000;    // 0xfffcf2c0 -> Truncated to 0xf2c0 (-3392) in 16-bit.
        outputBuffer[0] = asuint((int)a);
        outputBuffer[1] = asuint((int)b);
    }
}
