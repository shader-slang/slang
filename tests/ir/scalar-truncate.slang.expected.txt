7530
FFFFF2C0
0
0
